### DISCUSSION TITLE
v0.16.0

### DISCUSSION THREAD

**Discussion by moisout** [ViewTube/viewtube#2820] [Announcements]
# 👉 Highlights
> This release is quite significant, so I've decided to provide a detailed release post.

- ## 📺 New video player

  *Implemented by me (@moisout)*

  The video player has been rewritten. The old one was getting unmaintainable, most of the logic lived in a single file with over 1000 lines of code.
  
  The new player has now reached approximate feature parity with the old one, so I feel comfortable releasing it. I am not done working on it, you can check out the [roadmap](https://github.com/ViewTube/viewtube/issues/2387).

  Here are some of the new things it does:

  ### 🔭 Reliable quality switching
  All quality options delivered by YouTube are available in the quality switcher. This includes up to 8k, 60fps and HDR.
  
  The player will try to play up to the quality set in the users settings. You can also manually select a quality from the quality switcher.

  ![quality switcher](https://github.com/ViewTube/viewtube/assets/15955392/a5ce06fe-d4f8-4fcb-9589-b34ba63b3b57)

  ### 🗣️ Support for multiple language tracks
  If a video has multiple audio tracks, you can select them from the language switcher.

  ![language track switcher](https://github.com/ViewTube/viewtube/assets/15955392/97220e26-09c9-4668-b005-622ead0ebdd1)

  ### 📝 Subtitles
  Also, support for subtitles! The renderer for subtitles is pretty basic, so it doesn't support all the fancy stuff YouTube does, but it should work for most videos.

  ![subtitles example](https://github.com/ViewTube/viewtube/assets/15955392/da2a54a3-7dc0-4fae-a8cf-2cbd3b344af3)

  ### 🖼️ Seekbar hover preview
  Hovering over the seekbar now provides a frame preview, including the current SponsorBlock segment and chapter.

  ![seekbar hover preview example](https://github.com/ViewTube/viewtube/assets/15955392/f92d3739-d4c6-4789-9383-51c308e668c5)

- ## ⌚ Configurable subscription interval time

  *Implemented by @patricgruber* 

  You can now customize the interval for how often subscriptions should be updated. Configure the environment variable in minutes.
  ```env
  VIEWTUBE_SUBSCRIPTION_INTERVAL_TIME=60
  ```

- ## 🩳 Exclude shorts from search results
  *Implemented by @frozenduck*

  Exclude shorts from search results by enabling the option "Hide shorts" in the settings.

- ## 📜 Expandable subscriptions on homepage
  *Implemented by @nargacu83*

  Subscriptions on the homepage now work like the trending section. You can expand them to see more videos.

  ![subscriptions on homepage](https://github.com/ViewTube/viewtube/assets/15955392/d561e30d-3a1c-4751-9a31-ea4b29d07274)

- ## 🛑 Ability to set custom SponsorBlock API URL
  *Implemented by @patricgruber*

  You can now set a custom SponsorBlock API URL in the settings.

  ![sponsorblock API URL settings](https://github.com/ViewTube/viewtube/assets/15955392/9d0edd8b-fc2a-4883-9b6a-f5d988d216e0)

# Full Changelog

### Added

- Set custom SponsorBlock API URL (Thanks @patricgruber) [#2799](https://github.com/ViewTube/viewtube/pull/2799)
- Configurable subscription interval time (Thanks @patricgruber) [#2798](https://github.com/ViewTube/viewtube/pull/2798)
- Add option to exclude shorts from search results (Thanks @frozenduck) [#2795](https://github.com/ViewTube/viewtube/pull/2795)
- Expandable subscriptions on homepage (Thanks @nargacu83) [#2790](https://github.com/ViewTube/viewtube/pull/2790)
- Add option to select playback speed from a list (Thanks @frozenduck) [#2771](https://github.com/ViewTube/viewtube/pull/2771)
- New video player [#2388](https://github.com/ViewTube/viewtube/pull/2388)

### Fixed

- Rewrite comments API [#2772](https://github.com/ViewTube/viewtube/pull/2772)
- Perform search when clicking on autocomplete suggestion (Thanks @frozenduck) [#2770](https://github.com/ViewTube/viewtube/pull/2770)

<hr /><em>This discussion was created from the release <a href='https://github.com/ViewTube/viewtube/releases/tag/v0.16.0'>v0.16.0</a>.</em>

**Comment by n-ce** [comment]
Congratulations on the new release!

> The video player has been rewritten. The old one was getting unmaintainable, most of the logic lived in a single file with over 1000 lines of code.

That must have been a painful experience.

> 🩳 Exclude shorts from search results

😂

> 📝 Subtitles
Also, support for subtitles! The renderer for subtitles is pretty basic, so it doesn't support all the fancy stuff YouTube does, but it should work for most videos.

Is it TTML or a VTT renderer?

**Reply by moisout** [reply]
> Is it TTML or a VTT renderer?  

Neither, it just gets the transcript json from YouTube and displays it using the timestamps xD.
I'll have to find a way to get the real subtitles, there are some good subtitle rendering libraries out there.
